src/test/java/il/ac/bgu/cs/bp/bpjs/analysis/SnapshotBenchmarks.java

```diff
@@ -0,0 +1,361 @@
+        static int INITIAL_ARRAY_SIZE = 1;
+        static int ARRAY_STEP = 5; //Size of each step, how many objects are we adding
+        static int NUM_STEPS = 10; // How many steps to take
+
+
+        static void benchmarkVariableSizedArrays() throws Exception {
+            BenchmarkResult integerResults = measureIntegerSizes();
+
+            BenchmarkResult objectResults = measureObjectSize();
+            outputBenchResults(integerResults);
+            outputBenchResults(objectResults);
+        }
+
+        private static BenchmarkResult measureIntegerSizes() throws Exception {
+            System.out.println("Measuring effect of integer size");
+            return measureProgram(0);
+        }
+
+        private static BenchmarkResult measureObjectSize() throws Exception {
+            System.out.println("Measuring effect of object size");
+            return measureProgram(1);
+        }
+
+        private static BenchmarkResult measureProgram(int object_type) throws Exception {
+            VisitedStateStore store = new BThreadSnapshotVisitedStateStore();
+            VisitedStateStore storeHash = new HashVisitedStateStore();
+            DfsBProgramVerifier verifier = new DfsBProgramVerifier();
+            String testName = TEST_NAME + ((object_type == 1) ? "object" : "integer");
+            /*
+                Test for variable num_steps
+                we want to see if there's a non linear increase in snapshot size
+             */
+            int[][] snapshotSet = new int[ITERATIONS][];
+            long[][] verificationTimes = new long[ITERATIONS][];
+            long[][] verificationTimesHash = new long[ITERATIONS][];
+            BProgram[] programs = new BProgram[ITERATIONS];
+            for (int i = 0; i < ITERATIONS; i++) {
+                System.out.printf("Measuring for array step of %d size\n", ARRAY_STEP + i);
+                Map<String, Object> valueMap = new HashMap<>();
+                valueMap.put("INITIAL_ARRAY_SIZE", INITIAL_ARRAY_SIZE);
+                valueMap.put("ARRAY_STEP", ARRAY_STEP + i);
+                valueMap.put("NUM_STEPS", NUM_STEPS);
+                valueMap.put("OBJECT_TYPE", object_type);
+
+                BProgram programToTest = makeBProgram(IMPLEMENTATION, valueMap);
+                programs[i] = programToTest;
+                //Cleanup GC before and after, opportunistic
+                //have to clone the object because BPrograms are not reusable
+                System.gc();
+                snapshotSet[i] = getStateSizes(makeBProgram(IMPLEMENTATION, valueMap));
+                System.gc();
+                verifier.setVisitedNodeStore(store);
+                verificationTimes[i] = getVerificationTime(verifier, IMPLEMENTATION, valueMap, ITERATIONS);
+                verifier.setVisitedNodeStore(storeHash);
+                verificationTimesHash[i] = getVerificationTime(verifier, IMPLEMENTATION, valueMap, ITERATIONS);
+                System.gc();
```

Comment:
Not sure these `System.gc`s are needed. Note that the system does not even have to perform the garbage collection, it's just a recommendation.

Reply:
Agree, but otherwise we occasionally get junk data in results from GC running in middle of verification.

Reply:
Do you prefer we remove them?

Reply:
I think we can replace them with repetitions and removal of extreme values (so if the GC kicks in in 2 out fo 10 cases, we can remove that).
Or, we can decide that the GC is part of the Java platform, and we need to count it in. We can play with GC parameters and algorithms, as well as with memory adjustments.